src/frontend/src/btc/schedulers/btc-wallet.scheduler.ts

```diff
@@ -46,100 +65,159 @@ export class BtcWalletScheduler implements Scheduler<PostMessageDataRequestBtc>
 		});
 	}
 
-	private async loadBtcTransactionsData({ btcAddress }: { btcAddress: BtcAddress }): Promise<{
-		newTransactions: BitcoinTransaction[];
-		latestBitcoinBlockHeight: number | undefined;
-	}> {
+	private async loadBtcTransactionsData({
+		btcAddress
+	}: {
+		btcAddress: BtcAddress;
+	}): Promise<CertifiedData<BtcTransactionUi>[]> {
 		try {
 			const { txs: fetchedTransactions } = await btcAddressData({ btcAddress });
 
 			const newTransactions = fetchedTransactions.filter(({ hash }) =>
 				isNullish(this.store.transactions[`${hash}`])
 			);
 
-			this.store = {
-				...this.store,
-				transactions: {
-					...this.store.transactions,
-					...newTransactions.reduce(
-						(acc, transaction) => ({
-							...acc,
-							[transaction.hash]: {
-								certified: false,
-								data: transaction
-							}
-						}),
-						{}
-					)
-				}
-			};
-
 			const latestBitcoinBlockHeight = await btcLatestBlockHeight();
 
-			return { newTransactions, latestBitcoinBlockHeight };
+			return nonNullish(latestBitcoinBlockHeight)
+				? newTransactions.map((transaction) => ({
+						data: mapBtcTransaction({ transaction, btcAddress, latestBitcoinBlockHeight }),
+						certified: false
+					}))
+				: [];
 		} catch (error) {
 			// We don't want to disrupt the user experience if we can't fetch the transactions or latest block height.
 			console.error('Error fetching BTC transactions data:', error);
-			// TODO: Return an error instead of an object with empty array.
-			return { newTransactions: [], latestBitcoinBlockHeight: undefined };
+			// TODO: Return an error instead of an empty array.
+			return [];
 		}
 	}
 
-	private async loadBtcBalance({
+	private loadBtcBalance = async ({
 		identity,
-		bitcoinNetwork
-	}: {
-		identity: OptionIdentity;
-		bitcoinNetwork: BitcoinNetwork;
-	}): Promise<CertifiedData<bigint>> {
-		const balance = await getBtcBalance({
-			identity,
-			network: bitcoinNetwork,
-			minConfirmations: BTC_BALANCE_MIN_CONFIRMATIONS
-		});
-		const certifiedBalance = {
-			data: balance,
+		bitcoinNetwork,
+		btcAddress,
+		minterCanisterId,
+		certified = true
+	}: Omit<LoadBtcWalletParams, 'shouldFetchTransactions'>): Promise<
+		CertifiedData<bigint> | undefined
+	> => {
+		// Query BTC balance only if minterCanisterId is available
+		if (!certified) {
+			if (nonNullish(minterCanisterId) && isNullish(this.store.balance)) {
```

Comment:
Regarding `nonNullish(minterCanisterId)`, wouldn't it be more accurate to test if `nonNullish(minterCanisterId) && nonNullish(BITCOIN_CANISTER_IDS[minterCanisterId])` or we are 100% sure the minter canister id is always in the `BITCOIN_CANISTER_IDS`?